Sources/PenguinParallel/ThreadPool.swift

```diff
@@ -202,14 +164,32 @@ public struct InlineComputeThreadPool: TypedComputeThreadPool {
   /// Dispatch `fn` to be run at some point in the future (immediately).
   ///
   /// Note: this implementation just executes `fn` immediately.
-  public func dispatch(_ fn: (Self) -> Void) {
-    fn(self)
+  public func dispatch(_ fn: () -> Void) {
```

Comment:
```suggestion
  public func asynchronously(_ fn: () -> Void) {
```

Reply:
No completion handler or nothin'?   Seems a tad… limited.

Reply:
I didn't think you needed to add a completion handler at this level of abstraction. (Concretely, if you want to run something after `fn` completes, just wrap `fn` in a closure yourself, and put your code right there!) You can build up whatever you want on top, with whatever arbitrary synchronization primitives you'd like. (Concretely, I'm intentionally avoiding binding to any particular lock implementation.)